docs/plantuml/lease-shard-sync.puml

```diff
@@ -0,0 +1,63 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+
+autonumber
+
+title KCL Shard Syncing (Abridged)
+
+participant Scheduler as S
+participant LeaseCoordinator as LC
+participant PeriodShardSyncManager as PSS
+participant ShardSyncTask as SST
+participant "Lease Table (DDB)" as DDB
+participant LeaseRefresher as LR
+participant LeaseSynchronizer as LS
+participant HierarchicalShardSyncer as HSS
+participant ShardDetector as SD
+
+alt on initialization
+  S->S: create PeriodicShardSyncManager(\n  ..., leaseRefresher, leasesRecoveryAuditorExecutionFrequencyMillis, ...)
+  S->LC: initialize()
+  opt if lease table does not exist
+    LC->DDB: create lease table
+  end
+  S->PSS: start()
+  PSS->PSS: schedule self every\n  leasesRecoveryAuditorExecutionFrequencyMillis
+end
+
+loop every leasesRecoveryAuditorExecutionFrequencyMillis
+  opt if worker is not leader
+    PSS->PSS: go back to sleep
+  end
+
+  PSS->PSS: runShardSync()
+  opt if not required to sync shards
```

Comment:
Why would a leader not be required to sync shards? Is this if there are no shards, and if so can we say that explicitly

Reply:
Some examples:
1. KCL is a multi-threaded app, and the stream could be purged at some time before, or during, the `runShardSync()`
2. the known shard list is complete (i.e., no holes in the hash range)
3. the known shard list is incomplete, but the holes don't cross the threshold to trigger a sync
4. some future edit that may add, or remove, another condition that affects sync logic

> if so can we say that explicitly

The more specific/nuanced documentation is, the more likely it will suffer bitrot with an implementation change. We should balance documentation to convey ideas vs. repacking the implementation as English.

Update pending to provide some additional details, yet not a full enumeration.